src/subclassers.ts

```diff
@@ -121,16 +110,13 @@ export function WithResponse<
   };
 }
 
-export function WithFinalAuth<
-  TPrincipalKey extends string,
-  TAuthKey extends string
->(principalKey: TPrincipalKey, authorizerKey: TAuthKey) {
+export function WithFinalAuth<TProjector extends AnyAsyncProjector>(
+  projector: TProjector
+) {
   // tslint:disable-next-line:only-arrow-functions
   return function<
     TSuper extends Constructor<
```

Comment:
this type isn't right - in this case, TSuper doesn't have to be constrained cause we don't need anything from it.
Look up what Record<> does.
[ ] You probably just want to return function<TSuper> and not put any constraints on it.

Reply:
actually ... you DO want the superclass's instance ("this") to be compatible with the projector.  So here's try 2:
`return function<TSuper extends Parameter<TProjector>[0]>`
something like that
- [ ] Constrain the superclass to be compatible with the projector's parameter.